### ISSUE TITLE
[Feature Request] server blacklist

### ISSUE THREAD

**Issue by Byter09** [henrywhitaker3/Speedtest-Tracker#806]
I'm very happy with the tool so far and it helps me track when my ISP has problems and I can use it to leverage cost reduction with verifiable proof. That said, I sometimes have weird spikes in my graph and I manually went through all the tests and it turns out that some servers just don't seem to have the capacity in terms of bandwidth that I do, hence I get spikes indicating a 30% reduction in speed at random times.

My current solution was to just take a look at all tests and write down the good servers and use it in the whitelist. I have 4 servers in there, but this now makes me more reliable on those four servers operating at best performance at all times.

I'm proposing a black list in which I can enter server IDs of servers I do not want to use. This way I can keep the white list empty, wait for tests to come in and gradually remove the ones that cause spikes in the graph.

I hope this is possible to do in an easy manner, given that I know speedtest.net (CLI) only has a server parameter, but not an exclusion one.

Happy Holidays :)

**Comment by the-jeffski** [comment]
I second this request. Have just realised all the spikes I see are caused by it picking the same duff server .  I see the CLI lets you specify --exclude EXCLUDE  which is a server to ignore. A simple list of these or an option in the UI to blacklist the server would be ideal.

**Comment by cutzenfriend** [comment]
I also have this request. There is one server which gives me massive speed drops.

**Comment by WebBuddha** [comment]
+1

**Comment by GeHoGo** [comment]
+1

**Comment by hoccontrol** [comment]
+1 From me aswell! Same scenario as the others here!

**Comment by spencerandhill** [comment]
+1 Got the same issue with one server here.

**Comment by spcqike** [comment]
+1 

edit
Out of curiosity. Is it the same server for everyone? 
i have problems with speedtest.sewan-group.de

**Comment by a3rosol** [comment]
> +1
> 
> edit Out of curiosity. Is it the same server for everyone? i have problems with speedtest.sewan-group.de

That one seems oddly familiar. 
But seconding the request

**Comment by the-jeffski** [comment]
I've been looking at this a bit more and I see that the Ookla CLI doesn't actually support exclusions. So I see why this might not be possible now. I've written a shell script to run the cli and just added a skip loop to ignore the result if its a known bad server. Not ideal but stops me panicking the speed has dropped.

**Comment by Byter09** [comment]
@the-jeffski 
> I've been looking at this a bit more and I see that the Ookla CLI doesn't actually support exclusions.

Locally I have the following:
```
speedtest-cli 2.1.3
Python 3.10.2 (main, Jan 15 2022, 19:56:27) [GCC 11.1.0]
```
Haven't checked the docker image yet, but my version has a parameter called
```
--exclude EXCLUDE     Exclude a server from selection. Can be supplied multiple times
```

So.. looks like its fairly trivial to add.

EDIT:
Okay looking at the version in this project, I get
```
Speedtest by Ookla 1.0.0.2 (5ae238b) Linux/x86_64-linux-musl 5.15.18-1-MANJARO x86_64

The official command line client for testing the speed and performance
of your internet connection.
```
And I'm a bit confused. Is the `speedtest-cli` I have available on arch not the version being used in this speedtest tracker?

**Comment by the-jeffski** [comment]
@Byter09  That's right - the in the project is the official one from OOkla. The other CLI is the legacy one that tests over HTTP rather than web sockets. I find the older CLI maxes out around 600MB/s on my 1GB line whereas the official one handles the full speed.

**Comment by Byter09** [comment]
Well, another way to do this that I can think of then, would be to use the command to get a list of all servers, exclude the ones that the user doesn't want, and then explicitly use one from that remaining list. Is that possible?

**Comment by Byter09** [comment]
This is getting ridiculous at this point :D
![grafik](https://user-images.githubusercontent.com/16060697/154959204-03a04e70-1423-4068-aac3-01cb49f32f53.png)

**Comment by the-jeffski** [comment]
Your are not alone:   
![image](https://user-images.githubusercontent.com/63499439/154961137-33e56361-0773-4352-b29a-8415ded4b5a6.png)

Agree, your approach sounds the only option.  I've written a shell script now to run the CLI and do exactly that.

**Comment by Byter09** [comment]
Can you maybe make a merge request or so that triggers a new docker image being built with that in it? :D

**Comment by the-jeffski** [comment]
Ah sorry, wasn't maybe clear - I've done this as a separate test as I run home assistant so push the data there to be displayed now.  I looked at the code but couldn't understand it enough to mod.

**Comment by Byter09** [comment]
Oh I see. Haha. Alright :D
